includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -2453,6 +2459,48 @@ private function remove_disallowed_descendants( DOMElement $node, $allowed_desce
 		}
 	}
 
+	/**
+	 * Loop through node's siblings and remove them.
+	 *
+	 * @param DOMElement $node      Node.
+	 * @param string     $spec_name Spec name.
+	 */
+	private function remove_disallowed_siblings( DOMElement $node, $spec_name ) {
+		if ( ! $node->previousSibling || ! $node->nextSibling ) {
+			return;
+		}
+
+		$prev_sibling = $node->previousSibling;
+		while ( null !== $prev_sibling ) {
+			if ( XML_COMMENT_NODE !== $prev_sibling->nodeType ) {
```

Comment:
Why `XML_COMMENT_NODE`? Shouldn't this rather be doing:

```suggestion
			if ( ! ( $prev_sibling instanceof Element ) ) {
```